src/proto/h1/conn.rs

```diff
@@ -151,7 +151,9 @@ where I: AsyncRead + AsyncWrite + Unpin,
         self.state.version = msg.head.version;
 
         if msg.decode == DecodedLength::ZERO {
-            debug_assert!(!msg.expect_continue, "expect-continue needs a body");
+            if log_enabled!(log::Level::Debug) && msg.expect_continue {
```

Comment:
The `log_enabled!` here doesn't seem necessary, since there's no expensive computation going on in the block.

Reply:
True... I guess with `max-log-level` features, it could then just be eliminated... But I guess LLVM would also be able to eliminate `if some_bool { /* nothing */ }`. 🤷‍♂

Reply:
Yeah empty if blocks will go away.